obci/devices/haptics/haptics_stim_control_peer.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# OpenBCI - framework for Brain-Computer Interfaces based on EEG signal
+# Project was initiated by Magdalena Michalska and Krzysztof Kulewski
+# as part of their MSc theses at the University of Warsaw.
+# Copyright (C) 2008-2009 Krzysztof Kulewski and Magdalena Michalska
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author:
+#      Marian Dovgialo <marian.dowgialo@gmail.com>
+#
+
+
+from multiplexer.multiplexer_constants import peers, types
+from obci.control.peer.configured_multiplexer_server import ConfiguredMultiplexerServer
+from obci.configs import settings, variables_pb2
+from obci.utils.openbci_logging import log_crash
+from obci.devices.haptics.HapticsControl import HapticStimulator
+import os, sys
+
+class HapticStimulatorControlPeer(ConfiguredMultiplexerServer):
+    '''Class to control sensory stimulation'''
+    @log_crash
+    def __init__(self, addresses):
+        super(HapticStimulatorControlPeer, self).__init__(addresses=addresses,
+                                          type=peers.HAPTICS_STIMULATOR)
+        ids = self.config.get_param("id").split(":")
+        vid, pid = [int(i, base=16) for i in ids]
+        self.stim = HapticStimulator(vid, pid)
+        self.ready()
+        self.logger.info("HapticController init finished!")
+    
+    def close(self):
+        del self.stim
+        
+    def handle_message(self, mxmsg):
+        '''Receives message HAPTIC_CONTROL_MESSAGE and sends it
+        to stimulation board.
+        
+        Message contains info of type::
+            
+                message Variable {
+                    required string key = 1;
+                    required string value = 2;
+                    }
+        
+        :param key: - 'S', 'B' or 'T' - Single haptic channel or
```

Comment:
Such things should be formatted by some autoformatting tool, not by hand, but that's a different story for another time.